### PR TITLE
Search Component - Replaced class by func. comp

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Refactor

**The following has been addressed in the PR:** #116

**Description:**

in order to use react hooks, in this PR I updated the Search component by changing it from a class to a functional component.

I have also turned off this lint rule: `@typescript-eslint/explicit-function-return-type`, because 
it is not really helpful. please do you agree with me?

### CODE REVIEW

File: src/components/Search/Search.test.tsx [hunk 1]
```diff
@@ -1,272 +1,147 @@
...
+import { render, fireEvent, waitForElement } from '../../utils/test-react-testing-library';
+import api from '../../utils/api';
 
 import Search from './Search';
 
-const SEARCH_FILE_PATH = './Search';
-const API_FILE_PATH = '../../utils/calls';
-const URL_FILE_PATH = '../../utils/url';
+/* eslint-disable verdaccio/jsx-spread */
+const ComponentToBeRendered: React.FC = () => (
```
Comment (juanpicado) [inline src/components/Search/Search.test.tsx]: `ComponentToBeRendered` :D ... I hope ..

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -1,139 +1,91 @@
...
-        suggestions={suggestions}
-        suggestionsError={error}
-        suggestionsLoaded={loaded}
-        suggestionsLoading={loading}
-        value={search}
-      />
-    );
-  }
+const Search: React.FC<RouteComponentProps> = ({ history }) => {
+  const [suggestions, setSuggestions] = React.useState([]);
```
Comment (juanpicado) [inline src/components/Search/Search.tsx]: No blocking, but in the other places we use the `useState` from the import.

File: src/components/Search/Search.tsx [hunk 3]
```diff
@@ -1,139 +1,91 @@
...
+  const [loading, setLoading] = React.useState(false);
+  const [requestList, setRequestList] = React.useState<Array<{ abort: () => void }>>([]);
 
   /**
    * Cancel all the requests which are in pending state.
    */
-  private cancelAllSearchRequests: cancelAllSearchRequests = () => {
-    this.requestList.forEach(request => request.abort());
-    this.requestList = [];
+  const cancelAllSearchRequests = () => {
```
Comment (juanpicado) [inline src/components/Search/Search.tsx]: `useCallback` if you consider works here

Comment (juanpicado) [reply src/components/Search/Search.tsx]: Also below

Comment (priscilawebdev) [reply src/components/Search/Search.tsx]: Thank you for the suggestion :-) I checked, but  unfortunately in this case the callback doesn't help, it just adds more complexity

**Pull request merged**